app/controllers/sessions_controller.rb

```diff
@@ -3,28 +3,63 @@ def login_form
   end
 
   def login
-    username = params[:username]
-    if username and user = User.find_by(username: username)
-      session[:user_id] = user.id
-      flash[:status] = :success
-      flash[:result_text] = "Successfully logged in as existing user #{user.username}"
-    else
-      user = User.new(username: username)
-      if user.save
-        session[:user_id] = user.id
-        flash[:status] = :success
-        flash[:result_text] = "Successfully created new user #{user.username} with ID #{user.id}"
+   username = params[:username]
+   if username and user = User.find_by(username: username)
+     session[:user_id] = user.id
+     flash[:status] = :success
+     flash[:result_text] = "Successfully logged in as existing user #{user.username}"
+   else
+     user = User.new(username: username)
+     if user.save
+       session[:user_id] = user.id
+       flash[:status] = :success
+       flash[:result_text] = "Successfully created new user #{user.username} with ID #{user.id}"
+     else
+       flash.now[:status] = :failure
+       flash.now[:result_text] = "Could not log in"
+       flash.now[:messages] = user.errors.messages
+       render "login_form", status: :bad_request
+       return
+     end
+   end
+   redirect_to root_path
+ end
+
+
+  def create
```

Comment:
You should only need this controller action once you implement oauth - not the old `login` one